File: script-js/quickSort.js
function quickSort(arr) {
  // Cas de base : le tableau est déjà trié s'il est vide ou a un seul élément
  if (arr.length <= 1) {
    return arr;
  }

  // Choix du pivot comme premier élément du tableau
  const pivot = arr[0];

  // Partitionnement des éléments autour du pivot
  const left = [];
  const right = [];

  for (let i = 1; i < arr.length; i++) {
    // Les éléments inférieurs vont dans le tableau left, les autres dans le tableau right
    if (arr[i] < pivot) {
      left.push(arr[i]);
    } else {
      right.push(arr[i]);
    }
  }

  // Appel récursif sur les tableaux left et right, avec concaténation du pivot
  return [...quickSort(left), pivot, ...quickSort(right)];
}

// Exemple d'utilisation
const arrays = [1, 3, 5, 7, 2, 4, 6, 8, -9, -7];
console.log("Tableau initial :", arrays);
console.log("Tableau trié :", quickSort(arrays));

File: script-py/quick_sort.py
def quick_sort(arr):
    # Cas de base : le tableau est déjà trié s'il est vide ou a un seul élément
    if len(arr) <= 1:
        return arr

    # Choix du pivot comme premier élément du tableau
    pivot = arr[0]

    # Partitionnement des éléments autour du pivot
    left = [x for x in arr[1:] if x < pivot]
    right = [x for x in arr[1:] if x >= pivot]

    # Appel récursif sur les tableaux left et right, avec concaténation du pivot
    return quick_sort(left) + [pivot] + quick_sort(right)

# Exemple d'utilisation
arrays = [1, 3, 5, 7, 2, 4, 6, 8, -9, -7]
print("Tableau initial:", arrays)
print("Tableau trié:", quick_sort(arrays))
